### PR TITLE
Mergeperformance

### PR DESCRIPTION
**Pull request by amouhoub**
In this PR, @Aminelouati and I update the schedulingImplementation to gain arround 20% in execution time for merge tasks.
The merge tasks, like all tasks need to get the results of all its parent tasks from the DB. This operation is quite costly, and was performed twice before this PR.
In this PR, we modify the scheduling implementation by calling UpdateVariables that updates all kinds of variables just before creating and executing (merge) tasks. We also updated the UpdateVariables to return the results of parent tasks that it fetches from the DB to use it right after in creating the task

### CODE REVIEW

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/task/internal/InternalTask.java [hunk 1]
```diff
@@ -1199,6 +1203,7 @@ public synchronized void updateVariables(SchedulingService schedulingService) {
 
             updatedVariables.putAll(getSystemVariables());
         }
+        return taskResults;
```
Comment (marcocast) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/task/internal/InternalTask.java]: You never add anything to this map?

Comment (amouhoub) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/task/internal/InternalTask.java]: si,,, just a few lines above,,, inside the loop....line 1194

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/TimedDoTaskAction.java [hunk 2]
```diff
@@ -86,14 +86,17 @@
...
 
+    private Map<TaskId, TaskResult> taskResults;
+
     /**
      * Create a new instance of TimedDoTaskAction
      *
      * @param launcher the launcher of the task
+     * @param taskResults parent tasks results
      */
     public TimedDoTaskAction(InternalJob job, TaskDescriptor taskDescriptor, TaskLauncher launcher,
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/TimedDoTaskAction.java]: MAJOR Constructor has 8 parameters, which is greater than 7 authorized. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00107)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 8 issues
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 5 minor

Watch the comments in this conversation to review them.

#### 7 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [InternalTask.java#L382](https://github.com/ow2-proactive/scheduling/blob/e6fe4c9839b0b45e7d318dd52e32aec614597f36/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/task/internal/InternalTask.java#L382): A "NullPointerException" could be thrown; "parent" is nullable here. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)
1. ![MAJOR][MAJOR] [InternalTask.java#L1155](https://github.com/ow2-proactive/scheduling/blob/e6fe4c9839b0b45e7d318dd52e32aec614597f36/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/task/internal/InternalTask.java#L1155): A "NullPointerException" could be thrown; "obj" is nullable here. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)
1. ![MINOR][MINOR] [TimedDoTaskAction.java#L32](https://github.com/ow2-proactive/scheduling/blob/e6fe4c9839b0b45e7d318dd52e32aec614597f36/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/TimedDoTaskAction.java#L32): Remove this unused import 'java.util.ArrayList'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [TimedDoTaskAction.java#L33](https://github.com/ow2-proactive/scheduling/blob/e6fe4c9839b0b45e7d318dd52e32aec614597f36/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/TimedDoTaskAction.java#L33): Remove this unused import 'java.util.HashMap'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [TimedDoTaskAction.java#L35](https://github.com/ow2-proactive/scheduling/blob/e6fe4c9839b0b45e7d318dd52e32aec614597f36/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/TimedDoTaskAction.java#L35): Remove this unused import 'java.util.List'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [TimedDoTaskAction.java#L39](https://github.com/ow2-proactive/scheduling/blob/e6fe4c9839b0b45e7d318dd52e32aec614597f36/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/TimedDoTaskAction.java#L39): Remove this unused import 'org.apache.commons.collections4.ListUtils'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [TimedDoTaskAction.java#L50](https://github.com/ow2-proactive/scheduling/blob/e6fe4c9839b0b45e7d318dd52e32aec614597f36/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/TimedDoTaskAction.java#L50): Remove this unused import 'org.ow2.proactive.scheduler.core.properties.PASchedulerProperties'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request closed without merging**